src/core/coap/coap.cpp

```diff
@@ -76,11 +76,11 @@ void CoapBase::ClearRequests(const Ip6::Address *aAddress)
 {
     Message *nextMessage;
 
-    for (Message *message = static_cast<Message *>(mPendingRequests.GetHead()); message != NULL; message = nextMessage)
+    for (Message *message = mPendingRequests.GetHead(); message != NULL; message = nextMessage)
     {
         Metadata metadata;
 
-        nextMessage = static_cast<Message *>(message->GetNext());
+        nextMessage = MessageQueue::GetNext(*message);
```

Comment:
I went for a static method of `Coap::MessageQueue` to get the next entry from a previous method instead of overloading the `GetNext()` of `Coap::Message` (since a CoAP message can still be added in an `ot::MessageQueue` along with other `ot::Message` instances). 

Thoughts? Suggestions?

Reply:
Add method `Coap::GetNextCoap` that returns `Coap::Message` ?

Reply:
Added `Coap::Message::GetNextCoapMessage()`.